source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -110,7 +110,7 @@ AWSLambdaFilter::decodeHeaders(Http::RequestHeaderMap &headers,
       ENVOY_LOG(trace, "{}: stopping iteration to wait for STS credentials",
                 __func__);
       stopped_ = true;
-      return Http::FilterHeadersStatus::StopAllIterationAndBuffer;
+      return Http::FilterHeadersStatus::StopIteration;
```

Comment:
this is the bug fix. we have a filter that adds a body on header-only initial request and thus router filter gets called without appending the required credentials

this is an upstream envoy bug, as with `StopAllIterationAndBuffer` we would expect lambda filter `decodeHeaders` -> `decodeData` -> router filter `encodeHeaders`, but we see that `decodeData` is skipped and thus creds are missing (it races async STS call)

the filter is written in such a way that `StopAllIterationAndBuffer` or `StopIteration` can be used, thus the easier fix is to switch the use here (rather than fix upstream)

Reply:
fyi @nfuden